gormsupport/lifecycle_hard_delete.go

```diff
@@ -0,0 +1,29 @@
+package gormsupport
+
+import (
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/convert"
+	"github.com/jinzhu/gorm"
+)
+
+// LifecycleHardDelete struct contains all the items from gorm.Model except the ID and DeletedAt field,
+// hence we can embed the LifecycleHardDelete struct into Models that needs hard delete and alike.
+type LifecycleHardDelete struct {
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
```

Comment:
`LifecycleHardDelete` not used else where in this PR, hope we will be using this in future code soon.